### PR TITLE
Refactor particle params/state to use Pie

### PR DESCRIPTION
**Pull request by sethrj**
Instead of accessing ParticleDef directly, this now (like MaterialParams) provides a `ParticleView` with properties that can be accessed.

### CODE REVIEW

**Comment by sethrj** [comment]
@amandalund I could separate the fixes to the host interactor/cmake stuff into a separate PR if you'd like. I just did it here because it helped me fix the particle params/state issues locally (i.e. not requiring me to build on a cuda-enabled machines 😄 ).

File: src/physics/base/ParticleInterface.hh [hunk 1]
```diff
@@ -86,19 +108,47 @@ struct ParticleTrackState
...
 
+#ifndef __CUDA_ARCH__
+//---------------------------------------------------------------------------//
+/*!
+ * Resize particle states in host code.
+ */
+template<MemSpace M>
+inline void
+resize(ParticleStateData<Ownership::value, M>* data,
+       const ParticleParamsData<Ownership::const_reference, MemSpace::host>&,
```
Comment (amandalund) [inline src/physics/base/ParticleInterface.hh:145]: Unused? Will we need this in the future?

**Review by amandalund — APPROVED** [review]
Looking good!

File: src/physics/base/ParticleInterface.hh (hunk 1 above)
Comment (sethrj) [reply src/physics/base/ParticleInterface.hh:145]: Ah, this unused parameter is for consistency with the material resizer (and upcoming physics resizer). Conceivably the state initialization could require some runtime parameter data besides the number of tracks...

**Comment by sethrj** [comment]
Thanks for the review @amandalund !

**Pull request merged**